docs/writing-easyconfig-files.md

```diff
@@ -1,945 +1,994 @@
-********************************************
+- 32-character strings are considered to be MD5 checksums (`md5`)
+- 64-character strings are considered to be SHA256 checksums
+    (`sha256`)
+- (other lengths will result in an error message)
 
-If the easyconfig file does not specify any checksums yet, they are simply injected after the
-``sources`` (or ``patches``, if present) specification when ``--inject-checksums`` is used.
+The intention is to move towards making `sha256` the recommended and
+default checksum type.
 
-For example::
+Other checksum types are also supported: `adler32`, `crc32`, `sha1`,
+`sha512`, `size` (filesize in bytes). To provide checksum values of a
+specific type, elements of the `checksums` list can also be 2-element
+tuples of the form `('<checksum type>', '<checksum value>')`. For
+example:
 
-    $ eb bzip2-1.0.6.eb --inject-checksums
-    == temporary log file in case of crash /tmp/eb-Vm6w3e/easybuild-cAVQl6.log
-    == injecting sha256 checksums in /example/bzip2-1.0.6.eb
-    == fetching sources & patches for bzip2-1.0.6.eb...
-    == backup of easyconfig file saved to /example/bzip2-1.0.6.eb.bak_20170824200906...
-    == injecting sha256 checksums for sources & patches in bzip2-1.0.6.eb...
-    == * bzip2-1.0.6.tar.gz: a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd
-    == Temporary log file(s) /tmp/eb-Vm6w3e/easybuild-cAVQl6.log* have been removed.
-    == Temporary directory /tmp/eb-Vm6w3e has been removed.
+``` python
+checksums = [('sha512', 'f962008105639f58e9a4455c8057933ab0a5e2f43db8340ae1e1afe6dc2d24105bfca3b2e1f79cb242495ca4eb363c9820d8cea6084df9d62c4c3e5211d99266')]
+```
 
-The backup easyconfig file can be used to double-check the difference between the original easyconfig file
-and the one produced by ``--inject-checksums``::
+##### Adding or replacing checksums using `--inject-checksums` {: #inject_checksums }
 
-    $ diff -u /example/bzip2-1.0.6.eb.bak_20170824200906 /example/bzip2-1.0.6.eb
-    diff --git a//example/bzip2-1.0.6.eb.bak_20170824200906 b/example/bzip2-1.0.6.eb
-    index 46b2debed..2eb73f15a 100644
-    --- a/example/bzip2-1.0.6.eb.bak_20170824200906
-    +++ b/example/bzip2-1.0.6.eb
-    @@ -9,8 +9,9 @@ compressors), whilst being around twice as fast at compression and six times fas
-     toolchain = SYSTEM
-     toolchainopts = {'pic': True}
+Using the `--inject-checksums` command line option, you can let
+EasyBuild add or update checksums in one or more easyconfig files (which
+is significantly more convenient than doing it manually).
 
-    -sources = [SOURCE_TAR_GZ]
-     source_urls = ['http://www.bzip.org/%(version)s/']
-    +sources = [SOURCE_TAR_GZ]
-    +checksums = ['a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd']
+With `--inject-checksums`, checksums are injected for all sources and
+patches (if any), as well as for all sources & patches of every
+extension listed in `exts_list` (if any, see
+[module_extensions][module_extensions]).
```

Comment:
```suggestion
[Extensions][module_extensions]).
```

Based on some checking of old commits, this looks to be where it pointed in the past.